conda/resolve.py

```diff
@@ -764,22 +759,32 @@ def bad_installed(self, installed, new_specs):
         except TypeError:
             log.debug('Package set caused an unexpected error, assuming a conflict')
             solution = None
-        limit = None
+        limit = xtra = None
         if not solution or xtra:
             def get_(name, snames):
                 if name not in snames:
                     snames.add(name)
                     for fn in self.groups.get(name, []):
                         for ms in self.ms_depends(fn):
                             get_(ms.name, snames)
+            # New addition: find the largest set of installed packages that
+            # are consistent with each other, and include those in the
+            # list of packages to maintain consistency with
             snames = set()
+            try:
+                eq_optional_c = r2.generate_removal_count(C, specs)
+                solution, _ = C.minimize(eq_optional_c, C.sat())
+                snames.update(dists[Dist(q)]['name']
+                              for q in (C.from_index(s) for s in solution)
+                              if q and q[0] != '!' and '@' not in q)
+            except:
+                pass
```

Comment:
Frankly, unspecified pycosat errors. Note that in the absolute worst case, if such an exception were to occur, it would drop down to the exact behavior currently in `bad_installed`. I don't mind if we remove the exception handler.